camera/rpi-files/uploader.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+"""Data upload module.
+
+Uploads data to API endpoint.
+"""
+
+from datetime import datetime
+
+from data_batcher import Batch
+from events import EventType, create_event
+from openapi_client import ApiClient, CameraEvent, CamerasApi, Configuration
+from openapi_client.rest import ApiException
+
+
+class Uploader(object):
+    """Uploads events to API endpoint."""
+
+    def __init__(self, device_id: int, configuration: Configuration):
+        """Save path for writing.
+
+        Args:
+            device_id: device ID for writing
+            configuration: API client configuration
+        """
+        self._device_id = device_id
+        self._api_client = ApiClient(configuration)
+        self._api = CamerasApi(self._api_client)
+
+    def start(self, timestamp):
+        """Notify endpoint that upload starts.
+
+        Args:
+            timestamp: start time
+        """
+        event = create_event(EventType.monitoring_started, timestamp)
+        self._send_single_event(event)
+
+    def send(self, batch: Batch) -> bool:
+        """Send batch of events to the server.
+
+        Args:
+            batch: batch of events
+
+        Returns:
+            true when all uploads finalized
+        """
+        return self._send_all_as(
+            batch.entering, EventType.person_entered,
+        ) and self._send_all_as(batch.leaving, EventType.person_left)
+
+    def end(self, timestamp):
+        """Notify endpoint that upload ends.
+
+        Args:
+            timestamp: end time
+        """
+        event = create_event(EventType.monitoring_ended, timestamp)
+        self._send_single_event(event)
+        self._api_client.close()
+
+    def _send_single_event(self, event: CameraEvent) -> bool:
+        """Send single CameraEvent to endpoint.
+
+        Args:
+            event: CameraEvent to send
+
+        Raises:
+            RuntimeError: on upload failure
+
+        Returns:
+            HTTP response object
+        """
+        camera_pk = str(self._device_id)
+        try:
+            self._api.cameras_events_create(camera_pk, event)
+        except ApiException as ex:
+            msg = 'Could not upload event: {0}'.format(ex)
```

Comment:
rremember of fstrings, it jsut get's simpler in a way f'Could not upload event: {ex}'

Reply:
`fstring`s are forbidden per used style guide. See [wemake-python-styleguide#32](https://github.com/wemake-services/wemake-python-styleguide/issues/32) for discussion.